lib/src/authentication/sign_in/sign_in_view.dart

```diff
@@ -0,0 +1,143 @@
+import "package:kana_to_kanji/src/authentication/landing_view.dart";
+import "package:kana_to_kanji/src/authentication/sign_in/sign_in_view_model.dart";
+import "package:kana_to_kanji/src/authentication/widgets/button_apple_logo.dart";
+import "package:kana_to_kanji/src/authentication/widgets/button_google_logo.dart";
+import "package:kana_to_kanji/src/authentication/widgets/input_email.dart";
+import "package:kana_to_kanji/src/authentication/widgets/input_password.dart";
+import "package:kana_to_kanji/src/core/widgets/app_scaffold.dart";
+import "package:stacked/stacked.dart";
+
+class SignInView extends StatefulWidget {
+  static const routeName = "/authentication/login";
+
+  const SignInView({
+    super.key,
+  });
+
+  @override
+  State<SignInView> createState() => _SignInViewState();
+}
+
+class _SignInViewState extends State<SignInView> {
+  bool passwordVisible = true;
+
+  @override
+  void initState() {
+    super.initState();
+    passwordVisible = true;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final AppLocalizations l10n = AppLocalizations.of(context);
+
+    final Widget textDivider = Expanded(
+      child: Padding(
+        padding: const EdgeInsets.only(left: 20.0, right: 20.0),
+        child: Divider(
+          color: Theme.of(context).colorScheme.primaryContainer,
+        ),
+      ),
+    );
+
+    return ViewModelBuilder<SignInViewModel>.reactive(
+      viewModelBuilder: () => SignInViewModel(GoRouter.of(context)),
+      builder: (context, viewModel, child) => AppScaffold(
+        appBar: AppBar(
+          backgroundColor: Colors.transparent,
+          leading: InkWell(
+            key: const Key("sign_in_view_go_back_icon"),
+            borderRadius: BorderRadius.circular(25),
+            hoverColor: Theme.of(context).colorScheme.secondary,
+            splashColor: Theme.of(context).colorScheme.secondary,
+            focusColor: Theme.of(context).colorScheme.secondary,
+            highlightColor: Colors.transparent,
+            onTap: () => GoRouter.of(context).go(LandingView.routeName),
+            child: const SizedBox(
+              width: 36,
+              height: kToolbarHeight,
+              child: Icon(Icons.arrow_back_rounded),
+            ),
+          ),
+          toolbarHeight: kToolbarHeight,
+        ),
+        body: Form(
+          key: viewModel.formKey,
+          child: Padding(
+            padding: const EdgeInsets.only(top: 8.0),
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                InputEmail(controller: viewModel.emailController),
+                InputPassword(controller: viewModel.passwordController),
+                SizedBox(
+                  width: double.infinity,
+                  child: Padding(
+                    padding: const EdgeInsets.only(
+                        right: 16.0, top: 8.0, bottom: 8.0),
+                    child: InkWell(
```

Comment:
Here you can use a `TextButton` instead of an `InkWeel`

Reply:
Corrigé